src/components/Header.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useCallback } from "react";
+import styled from "styled-components";
+import {
+  space,
+  typography,
+  layout,
+  flexbox,
+  grid,
+  position,
+  SpaceProps,
+  TypographyProps,
+  LayoutProps,
+  FlexboxProps,
+  GridProps,
+  PositionProps,
+} from "styled-system";
+import NavMenu from "./NavMenu";
+import i18next from "i18next";
+import { NavLink } from "react-router-dom";
+
+const HeaderContainer = styled.div<
+  LayoutProps & SpaceProps & GridProps & PositionProps & FlexboxProps
+>`
+  width: 100%;
+  ${layout};
+  ${space};
+  ${grid};
+  ${position};
+  ${flexbox};
+`;
+
+const H1 = styled.h1<SpaceProps & TypographyProps & GridProps>`
+  ${space};
+  ${typography};
+  ${grid};
+`;
+
+const Container = styled.div<
+  FlexboxProps & GridProps & LayoutProps & SpaceProps & TypographyProps
+>`
+  background-color: black;
+  width: 100%;
+  ${flexbox};
+  ${grid};
+  ${layout};
+  ${space};
+  ${typography};
+`;
+
+const LangButton = styled.button<SpaceProps & TypographyProps>`
+  background: transparent;
+  border: none;
+  ${typography};
+  ${space};
+  transition: transform 0.2s;
+  transform-origin: left;
+  transform-origin: right;
+  &:hover {
+    transform: scale(1.05);
+`;
+
+const Header = () => {
+  const fontSizes = [0, null, null, null, null, 1, 4, null, null, 5];
```

Comment:
this seems execessive, is there a shorthand way to do this?

Reply:
yeah it's ugly but genuinely don't know the answer to that. getting the fonts right on every breakpoint was tricky and this is a product of trial and error. if I stumble across a better way of doing this, I'll for sure come back to this with a refactor. otherwise will probably leave as is purely because it works